lib/Runtime/ByteCode/OpCodes.h

```diff
@@ -827,6 +827,8 @@ MACRO_BACKEND_ONLY(     TrapIfMinIntOverNegOne, Reg3,       OpSideEffect)
 MACRO_BACKEND_ONLY(     TrapIfZero,         Reg3,           OpSideEffect)
 MACRO_BACKEND_ONLY(     TrapIfUnalignedAccess, Reg3,        OpSideEffect)
 
+MACRO_EXTEND_WMS(       SpreadObjectLiteral, Reg2,          OpSideEffect)
```

Comment:
I think this also needs OpHasImplicitCall to describe to the JIT that it might make reentrant calls to Javascript while executing (because it runs getters on the source object), and therefore might invalidate some assumptions about the current state. We should also have a unit test verifying that we appropriately bail out when necessary. For instance, maybe something like the following would fail without that flag:

```javascript
// command line flags: -bgjit- -maxinterpretcount:1 -off:simplejit
const obj = { a: 2 };
function f(x) {
    const a = obj.a;
    const unused = {...x};
    return a + obj.a;
}

// Train it that ...x is not reentrant, so it emits code that assumes the second obj.a matches the first
f({});

// Now call with a getter and verify that it bails out when the previous assumption is invalidated
const reentrantResult = f({ get b() { obj.a = 3; } });
assert.areEqual(5, reentrantResult);
```

If this test passes as written, it probably means the test needs modification, not that the opcode doesn't need OpHasImplicitCall.

Reply:
Makes sense.